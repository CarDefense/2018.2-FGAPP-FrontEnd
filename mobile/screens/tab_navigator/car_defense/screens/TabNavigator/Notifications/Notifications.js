import React, { Component } from 'react';
import {
    View,
    StyleSheet,
    ScrollView,
    TouchableOpacity,
    Image,
} from 'react-native';


export default class Notifications extends Component {

    PrivateNotifications = () => {
        const { state } = this.props.navigation;
        var user = state.params ? state.params.user : undefined;
        this.props.navigation.navigate('PrivateNotifications', { user: user })
    }

    PublicNotifications = () => {
        const { state } = this.props.navigation;
        var user = state.params ? state.params.user : undefined;
        this.props.navigation.navigate('PublicNotifications', { user: user })
    }


    render() {
        return (
        <View
        style={{ backgroundColor: '#00ACC1', flex: 1 }}
        >
            <ScrollView>
                <View style={styles.container}>
                    <TouchableOpacity
                        onPress={() => this.PrivateNotifications()}
                    >
                        <Image style={styles.avatar}
                            source={ require('../../../../../../images/oie_transparent.png') }
                        />
                    </TouchableOpacity>
                </View>
                <View style={styles.container1}>
                    <TouchableOpacity
                        onPress={() => this.PublicNotifications()}
                    >
                        <Image style={styles.avatar}
                            source={require('../../../../../../images/al.png')}
                        />
                    </TouchableOpacity>
                </View>
            </ScrollView>
            </View>
        );
    }
}

const styles = StyleSheet.create({
    container: {
        justifyContent: 'center',
        alignItems: 'center',
        paddingTop: 100
    },
    container1: {
        justifyContent: 'center',
        alignItems: 'center',
        padding: 50,
    },
    avatar: {
        width: 240,
        height: 240, 
        marginBottom: 10,
        
    },
    name: {
        fontSize: 18,
        color: "white",
        fontWeight: '800',
        justifyContent: "center",
        alignItems: "center"
    },
});
