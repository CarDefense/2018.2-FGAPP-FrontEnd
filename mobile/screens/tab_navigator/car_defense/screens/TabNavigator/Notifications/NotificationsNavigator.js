import React, { Component } from "react";
import {StackNavigator} from 'react-navigation'
import Notifications from './Notifications'


export default class NotificationsNavigator extends Component {

    render() {
        return (
            <NotificationsStackNavigator />
        );
    }
}

const NotificationsStackNavigator = new StackNavigator({

    Notificatios: {
        screen: Notifications,
        navigationOptions: ({ navigation }) => ({
            header: null,
        }),
    },
})
