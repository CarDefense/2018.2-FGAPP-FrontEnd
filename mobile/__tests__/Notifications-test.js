import 'react-native'
import React from 'react'
import renderer from 'react-test-renderer'
import Notifications from '../screens/tab_navigator/car_defense/screens/TabNavigator/Notifications/Notifications';

test('Notifications snapShot', () => {
    const snap = renderer.create(
        <Notifications/>
    ).toJSON()
    expect(snap).toMatchSnapshot()
})
