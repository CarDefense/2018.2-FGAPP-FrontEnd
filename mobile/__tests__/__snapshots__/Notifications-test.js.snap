// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Notifications snapShot 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#00ACC1",
      "flex": 1,
    }
  }
>
  <RCTScrollView>
    <View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "paddingTop": 100,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={1}
            style={
              Object {
                "height": 240,
                "marginBottom": 10,
                "width": 240,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "alignItems": "center",
            "justifyContent": "center",
            "padding": 50,
          }
        }
      >
        <View
          accessibilityComponentType={undefined}
          accessibilityLabel={undefined}
          accessibilityTraits={undefined}
          accessible={true}
          collapsable={undefined}
          hasTVPreferredFocus={undefined}
          hitSlop={undefined}
          isTVSelectable={true}
          nativeID={undefined}
          onLayout={undefined}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "opacity": 1,
            }
          }
          testID={undefined}
          tvParallaxProperties={undefined}
        >
          <Image
            source={1}
            style={
              Object {
                "height": 240,
                "marginBottom": 10,
                "width": 240,
              }
            }
          />
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
